debezium-connector-postgres/src/main/java/io/debezium/connector/postgresql/PostgresStreamingChangeEventSource.java

```diff
@@ -191,7 +198,13 @@ else if (message.getOperation() == Operation.COMMIT) {
                         pauseNoMessage.sleepWhen(true);
                     }
                 }
-                connection.commit();
+                if (!isInPreSnapshotCatchUpStreaming()) {
+                    // During catch up streaming, the streaming phase needs to hold a transaction open so that
+                    // teh phase can stream event up to a specific lsn and the snapshot that occurs after the catch up
```

Comment:
```suggestion
                    // the phase can stream event up to a specific lsn and the snapshot that occurs after the catch up
```

Reply:
Fixed in https://github.com/debezium/debezium/commit/3a9f502dc2243f02af8187a1ffb2e580df84d36c